src/CiaoService.ts

```diff
@@ -108,6 +108,15 @@ export interface ServiceOptions {
    * configured to ignore this interface, the service won't be advertised on the interface.
    */
   restrictedAddresses?: (InterfaceName | IPAddress)[];
+
+  /**
+   * If specified, the mDNS server will advertise this address only.
+   * This is to help in situations like Docker where the interface's IP address does not equal the external one.
+   * These value are merely here for the purpose of advertising, and does not carry any semantic
+   * value. It is up to the administrator to verify that the service is reachable on this IP address.
+   */
+  overrideAdvertised?: string;
```

Comment:
I think the option needs a small note/documentation that the `overrideAdvertised` address will be used for every interface the service is advertised on (which might be the case).

Reply:
I'll address this one later as I think your other comment / changes will impact this.